### PR TITLE
Diff calculation: improve performance

### PR DESCRIPTION
**Pull request by christianguenter2**
Previously we calculated the whole repo diff, which might be very expensive, and then filter by requested files.
The new logic ensures that only diffs for requested files are calculated.

### CODE REVIEW

File: src/ui/zcl_abapgit_gui_page_diff.clas.abap [hunk 1]
```diff
@@ -1020,4 +1000,19 @@ CLASS zcl_abapgit_gui_page_diff IMPLEMENTATION.
     ENDCASE.
 
   ENDMETHOD.
+
+
+  METHOD is_file_requested.
+
+    IF lines( it_files ) = 0.
```
Comment (larshp) [inline src/ui/zcl_abapgit_gui_page_diff.clas.abap:1007]: I guess this IF is a suboptimization? READ TABLE for an empty table _should_ be as fast as an IF

File: src/ui/zcl_abapgit_gui_page_diff.clas.abap [hunk 2]
```diff
@@ -1020,4 +1000,19 @@ CLASS zcl_abapgit_gui_page_diff IMPLEMENTATION.
...
+      rv_is_file_requested = abap_true.
+      RETURN.
+    ENDIF.
+
+    READ TABLE it_files WITH KEY file-filename = is_status-filename
```
Comment (larshp) [inline src/ui/zcl_abapgit_gui_page_diff.clas.abap:1012]: can `path` be added to the  conditions? the table has a unique key which will make the READ fast,

```abap
  TYPES:
    ty_stage_tt TYPE SORTED TABLE OF ty_stage
          WITH UNIQUE KEY file-path file-filename .
```

**Review by larshp — CHANGES_REQUESTED** [review]
see comments

**Comment by mbtools** [comment]
this...

```abap
LOOP AT lt_status ...
   READ it_files...
   IF sy-subrc = 0.
     append_diff( ...
   ENDIF.
ENDLOOP.
```

File: src/ui/zcl_abapgit_gui_page_diff.clas.abap (hunk 2 above)
Comment (christianguenter2) [reply src/ui/zcl_abapgit_gui_page_diff.clas.abap]: That's right, thanks for the hint. Updated.

File: src/ui/zcl_abapgit_gui_page_diff.clas.abap (hunk 1 above)
Comment (christianguenter2) [reply src/ui/zcl_abapgit_gui_page_diff.clas.abap:1017]: Well, wouldn't that change its semantics? READ TABLE for an empty table gives, if I'm not mistaken, SUBRC = 4 and then the method returns ABAP_FALSE.

**Comment by christianguenter2** [comment]
@larshp updated and commented
@mbtools I'm not sure whether I get your comment. Do you suggest to inline the method? Well that's what I had first, but if `IT_FILES` is empty than the diff should also be appended. Then the `IF` gets a bit convoluted therefore I introduced the method to add a bit more context.

File: src/ui/zcl_abapgit_gui_page_diff.clas.abap (hunk 1 above)
Comment (mbtools) [reply src/ui/zcl_abapgit_gui_page_diff.clas.abap:1017]: If it_files is empty, the original status loop was ok. If it's not empty, then I would do a loop with read table it_files inside.

Comment (christianguenter2) [reply src/ui/zcl_abapgit_gui_page_diff.clas.abap:1017]: Do you mean something like this?
```
      IF lines( it_files ) = 0.

        LOOP AT lt_status ASSIGNING <ls_status> WHERE match IS INITIAL.

          append_diff( it_remote = lt_remote
                       it_local  = lt_local
                       is_status = <ls_status> ).

        ENDLOOP.

      ELSE.

        LOOP AT lt_status ASSIGNING <ls_status> WHERE match IS INITIAL.

          READ TABLE it_files WITH KEY file-path     = <ls_status>-path
                                       file-filename = <ls_status>-filename
                              TRANSPORTING NO FIELDS.
          IF sy-subrc = 0.
            append_diff( it_remote = lt_remote
                         it_local  = lt_local
                         is_status = <ls_status> ).
          ENDIF.

        ENDLOOP.

      ENDIF.

```

Comment (mbtools) [reply src/ui/zcl_abapgit_gui_page_diff.clas.abap:1017]: Yes. Many ways to code the same thing but that looks good to me.

Comment (larshp) [reply src/ui/zcl_abapgit_gui_page_diff.clas.abap:1017]: ahh, okay, it says abap_true in line 1018

**Review by larshp — APPROVED** [review]
looks good to me, if there are minor code changes we can do it in a follow up PR, merging

**Pull request merged**